src/Homepage/Button/SearchButton.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { Component } from 'react';
+import './_search-button.scss';
+
+import { Link } from 'react-router-dom';
+
+
+
+export default class SearchButton extends Component {
+
+    shouldComponentUpdate(nextProps, nextState, nextContext) {
```

Comment:
You should consider using a type checking library. Let's take a look at `prop-types` library and use it on your project.